code-challenges.test.js

```diff
@@ -15,37 +15,108 @@
+
+// *****It mostly works as outlined in the Pseudo, went with sorting in the return statement since it seems MUCH more efficient than mapping a new array. I do wonder if shift is more efficient than splice, though. I assume so since the code has to do less 'thinking' with .shift which automatically targets the beginning of the array instead of whatever parameters you define with .splice. *****
+// *****Also served as a nice experiment with the sort method. I didn't know you could use sort with Math.random, that's pretty handy!*****
+ 
+
 
 // --------------------2) Create a function that takes in an object that contains up votes and down votes and returns the end tally.
 
+// Input: an object containing two numbers
+// Output: the two numbers combined.
+// Pseudo: Need to subtract the downvotes from the upvotes within the object and return the total.
+
 // a) Create a test with expect statements for each of the variables provided.
 
+describe("rockTheVote", () => {
+  it("takes in an object that contains up votes and down votes and returns the end tally.", () => {
+    expect(rockTheVote(votes1)).toEqual(11)
+    expect(rockTheVote(votes2)).toEqual(-31)
+  })
+});
+
+// [error 1:   ● Test suite failed to run --- ReferenceError: Cannot access 'rockTheVote' before initialization]  ***dangit, I'll get there eventually. forgot to 'stringify' the function name in the test...***
+// [error 2: TypeError: rockTheVote(...).toEqual is not a function] ***not a function my butt...misallocated some parantheticals.***
+
 const votes1 = { upVotes: 13, downVotes: 2 }
 // Expected output: 11
 const votes2 = { upVotes: 2, downVotes: 33 }
 // Expected output: -31
 
 // b) Create the function that makes the test pass.
+const rockTheVote = (input) => {return (input.upVotes - input.downVotes)};  // ***I promise I'll stop.***
+
+// *****this seems too easy not to be a trap, and now I'm paranoid that I missed something... I guess the only tricky part was remembering that we're working with objects so you need to access it in a different manner from an array?*****
 
 // --------------------3) Create a function that takes in two arrays as arguments and returns one array with no duplicate values. STRETCH: Use the spread operator to pass in a dynamic number of arguments.
 
+// input: two arrays containing strings
+// output: one array with no duplicate values
+// Pseudo: Concatinate the two input arrays, filter out the duplicates, and return the new filtered array.
+
 // a) Create a test with an expect statement using the variables provided.
+describe("hotGlue", () => {
+  it("takes in two arrays as arguments and returns one array with no duplicate values", () => {
+    expect(hotGlueTwoOhCrapItsEverywhere(dataArray1, dataArray2)).toEqual(["array", "object", "number", "string", "Boolean", "null", "undefined"])
```

Comment:
This passed the second test with a different function name in the expect? Usually will throw an error because your describe is going to correlate with your expect 🤔